packages/core/src/collections/notes.ts

```diff
@@ -265,6 +290,14 @@ export class Notes implements ICollection {
   favorite(state: boolean, ...ids: string[]) {
     return this.collection.update(ids, { favorite: state });
   }
+  async archive(state: boolean, ...ids: string[]) {
+    await this.collection.update(ids, { archived: state });
+    if (state) {
+      this.cache.archives.push(...ids);
+    } else {
+      deleteItems(this.cache.archives, ...ids);
+    }
```

Comment:
This assumes that the id isn't already in the cache which could be incorrect i.e. if `archive` is called multiple times with the same `state` value.